src/TabbedArea.js

```diff
@@ -1,167 +1,25 @@
   },
-
-  getInitialState() {
-    let defaultActiveKey = this.props.defaultActiveKey != null ?
-      this.props.defaultActiveKey : getDefaultActiveKeyFromChildren(this.props.children);
-
-    return {
-      activeKey: defaultActiveKey,
-      previousActiveKey: null
-    };
-  },
-
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.activeKey != null && nextProps.activeKey !== this.props.activeKey) {
-      // check if the 'previousActiveKey' child still exists
-      let previousActiveKey = this.props.activeKey;
-      React.Children.forEach(nextProps.children, (child) => {
-        if (React.isValidElement(child)) {
-          if (child.props.eventKey === previousActiveKey) {
-            this.setState({
-              previousActiveKey
-            });
-            return;
-          }
-        }
-      });
-
-      // if the 'previousActiveKey' child does not exist anymore
-      this.setState({
-        previousActiveKey: null
-      });
-    }
-  },
-
-  handlePaneAnimateOutEnd() {
-    this.setState({
-      previousActiveKey: null
-    });
-  },
-
   render() {
-    let { id, ...props } = this.props;
-
-    function renderTabIfSet(child) {
-      return child.props.tab != null ? this.renderTab(child) : null;
-    }
+    let {children, ...props} = this.props;
+    let tabTitles = [];
 
-    let nav = (
-      <Nav {...props} activeKey={this.getActiveKey()} onSelect={this.handleSelect} ref="tabs">
-        {ValidComponentChildren.map(this.props.children, renderTabIfSet, this)}
-      </Nav>
-    );
-
-    return (
-      <div>
-        {nav}
-        <div id={id} className="tab-content" ref="panes">
-          {ValidComponentChildren.map(this.props.children, this.renderPane)}
-        </div>
-      </div>
-    );
-  },
-
-  getActiveKey() {
-    return this.props.activeKey != null ? this.props.activeKey : this.state.activeKey;
-  },
-
-  renderPane(child, index) {
-    let previousActiveKey = this.state.previousActiveKey;
-
-    let shouldPaneBeSetActive = child.props.eventKey === this.getActiveKey();
-    let thereIsNoActivePane = previousActiveKey == null;
-
-    let paneIsAlreadyActive = previousActiveKey != null && child.props.eventKey === previousActiveKey;
-
-    return cloneElement(
-        child,
-        {
-          active: shouldPaneBeSetActive && (thereIsNoActivePane || !this.props.animation),
-          id: panelId(this.props, child),
-          'aria-labelledby': tabId(this.props, child),
-          key: child.key ? child.key : index,
-          animation: this.props.animation,
-          onAnimateOutEnd: paneIsAlreadyActive ? this.handlePaneAnimateOutEnd : null
-        }
-      );
-  },
-
-  renderTab(child) {
-    let {eventKey, className, tab, disabled } = child.props;
+    tabTitles = ValidComponentChildren.map(function(child) {
+      let {tab, ...others} = child.props;
+      tabTitles.push(<TabPane title={tab} {...others}/>);
```

Comment:
I believe this is a merged bug now :smile: 
@taion meant here https://github.com/pivotal-cf/react-bootstrap/commit/caeeabb0b8e597055ddad60c7962586fcb478092#commitcomment-12650327
to re-write this line: `tabTitles.push(<TabPane title={tab} {...others}/>);` too o/c.
into:
`return (<TabPane title={tab} {...others}/>);`